cmethods/__init__.py

```diff
@@ -1209,218 +395,190 @@ def quantile_delta_mapping(
-            if cls.nan_or_equal(value1=global_max, value2=global_min):
+            if nan_or_equal(value1=global_max, value2=global_min):
                 return simp
 
             wide = global_max / n_quantiles
             xbins = np.arange(global_min, global_max + wide, wide)
 
-            cdf_obs = cls.get_cdf(obs, xbins)
-            cdf_simh = cls.get_cdf(simh, xbins)
-            cdf_simp = cls.get_cdf(simp, xbins)
+            cdf_obs = get_cdf(obs, xbins)
+            cdf_simh = get_cdf(simh, xbins)
+            cdf_simp = get_cdf(simp, xbins)
 
             epsilon = np.interp(simp, xbins, cdf_simp)  # Eq. 1.1
-            QDM1 = cls.get_inverse_of_cdf(cdf_obs, epsilon, xbins)  # Eq. 1.2
+            QDM1 = get_inverse_of_cdf(cdf_obs, epsilon, xbins)  # Eq. 1.2
 
-            delta = cls.ensure_devidable(  # Eq. 2.3
-                simp, cls.get_inverse_of_cdf(cdf_simh, epsilon, xbins)
+            delta = ensure_devidable(  # Eq. 2.3
+                simp,
+                get_inverse_of_cdf(cdf_simh, epsilon, xbins),
+                max_scaling_factor=self.MAX_SCALING_FACTOR,
             )
             return QDM1 * delta  # Eq. 2.4
         raise NotImplementedError(
-            f"{kind} not available for quantile_delta_mapping. Use '+' or '*' instead."
+            f"{kind=} not available for quantile_delta_mapping. Use '+' or '*' instead."
         )
 
-    @classmethod
-    def check_types(
-        cls: CMethods,
-        obs: Union[list, np.ndarray, np.generic, xr.core.dataarray.DataArray],
-        simh: Union[list, np.ndarray, np.generic, xr.core.dataarray.DataArray],
-        simp: Union[list, np.ndarray, np.generic, xr.core.dataarray.DataArray],
-    ) -> None:
-        """
-        Checks if the parameters are in the correct type. **only used internally**
-        """
-        phrase: str = "must be type list, np.ndarray or xarray.core.dataarray.DataArray"
-        valid_types: tuple = (list, np.ndarray, np.generic, xr.core.dataarray.DataArray)
-
-        if not isinstance(obs, valid_types):
-            raise TypeError(f"'obs' {phrase}")
-        if not isinstance(simh, valid_types):
-            raise TypeError(f"'simh' {phrase}")
-        if not isinstance(simp, valid_types):
-            raise TypeError(f"'simp' {phrase}")
-
-    @staticmethod
-    def nan_or_equal(value1: float, value2: float) -> bool:
+    def adjust(
+        self: CMethods, method: str, obs: XRData, simh: XRData, simp: XRData, **kwargs
+    ) -> XRData:
         """
-        Returns True if the values are equal or at least one is NaN
-
-        :param value1: First value to check
-        :type value1: float
-        :param value2: Second value to check
-        :type value2: float
-        :return: If any value is NaN or values are equal
-        :rtype: bool
-        """
-        return np.isnan(value1) or np.isnan(value2) or value1 == value2
+        Function to apply a bias correction technique on 1-and 3-dimensional
```

Comment:
FYI this works generically on N-dimensional datasets, not just 1- and 3-. I tested on `('time')`, `('lat', 'time')`, `('lat', 'lon', 'time')`, and `('lat', 'lon', 'depth', 'time')` and it all works very fast.